3725/3/app/controllers/users_controller.rb

```diff
@@ -0,0 +1,37 @@
+class UsersController < ApplicationController
+  attr_reader :user
+
+  get '/login' do
+    erb :'users/log_in'
+  end
+
+  post '/login' do
+    @user = User.find_by(params[:email])
+    if @user.password == params[:password]
+      session[:user_id] = @user.id
+      redirect '/posts/all'
+    else
+      session[:error] = 'Log in failed!'
+      redirect '/users/login'
+    end
+  end
+
+  get '/signin' do
+    erb :'users/sign_in'
+  end
+
+  post '/signin' do
+    @user = User.new(params[:user])
+    if @user.save!
```

Comment:
`save!` will raise the exception for invalid user. Use `valid?` instead of it. Method `create` will return you object anyway so instead of new you could use `create`